monitoring-engine/src/main/java/io/jenkins/plugins/remotingopentelemetry/engine/PreMain.java

```diff
@@ -15,14 +17,24 @@ public static void premain(String agentArgs, Instrumentation inst) {
             return;
         }
 
+        String metricsFilterPatternString = System.getenv("REMOTING_OTEL_METRIC_FILTER");
+        Pattern metricsFilterPattern;
+        if (metricsFilterPatternString == null) metricsFilterPatternString = ".*";
```

Comment:
It might be more efficient to pass the null pattern and to handle it as separate case

Reply:
I see, but it does the pattern evaluation only once for each metric at the registration phase, and I couldn't come up with a better way to achieve this without redundant code. So I will merge this PR as it is and I'll fix this later if the efficiency seems to be a problem. Thank you for your review!